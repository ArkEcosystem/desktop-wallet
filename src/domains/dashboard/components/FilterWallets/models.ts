export type FilterWalletsHookProps = {
	networks?: any;
	viewType?: "list" | "grid";
	useTestNetworks?: boolean;
	walletsDisplayType?: string;
	selectedNetworkIds?: string[];
	isFilterChanged?: boolean;
};

export type FilterWalletsProps = {
	onChange?: (key: string, value: any) => void;
} & FilterWalletsHookProps;
