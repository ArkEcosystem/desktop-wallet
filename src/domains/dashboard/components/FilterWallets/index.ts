export * from "./FilterWallets";
export * from "./hooks";
export * from "./models";
