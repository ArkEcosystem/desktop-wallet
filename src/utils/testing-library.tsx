import { render } from "@testing-library/react";
import { i18n } from "app/i18n";
import { createMemoryHistory } from "history";
import React from "react";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";

const WithProviders = ({ children }: { children: React.ReactNode }) => {
	return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};

const customRender = (component: React.ReactElement, options: any) =>
	render(component, { wrapper: WithProviders, ...options });

const renderWithRouter = (
	component: React.ReactElement,
	{ routes = ["/"], history = createMemoryHistory({ initialEntries: routes }) } = {},
) => {
	const RouterWrapper = ({ children }: { children: React.ReactNode }) => (
		<Router history={history}>{children}</Router>
	);

	return {
		...customRender(component, { wrapper: RouterWrapper }),
		history,
	};
};

export * from "@testing-library/react";

export { customRender as render, renderWithRouter };
