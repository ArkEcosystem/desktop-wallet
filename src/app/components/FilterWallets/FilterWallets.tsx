import React, { useState } from "react";
import { SelectNetwork } from "../SelectNetwork";
import { Divider } from "app/components//Divider";
import { Badge } from "../Badge";
import { Toggle } from "../Toggle";

type FilterWalletsProps = {
	networks?: any;
	visiblePortfolioView?: any;
	visibleTransactionsView?: any;
	onNetworkChange?: any;
	onViewAllNetworks?: any;
	onWalletsDisplay?: any;
	togglePortfolioView?: any;
	toggleTransactionsView?: any;
};

export const FilterWallets = ({
	networks,
	onNetworkChange,
	onViewAllNetworks,
	onWalletsDisplay,
	togglePortfolioView,
	toggleTransactionsView,
	visibleTransactionsView,
	visiblePortfolioView,
}: FilterWalletsProps) => {
	const [showPortfolio, setShowPortfolio] = useState(visiblePortfolioView);
	const [showTransactions, setShowTransactions] = useState(visibleTransactionsView);

	const toggleTransactions = (isChecked: boolean) => {
		setShowTransactions(isChecked);
		if (typeof toggleTransactionsView === "function") toggleTransactionsView(isChecked);
	};

	const togglePortfolio = (isChecked: boolean) => {
		setShowPortfolio(isChecked);
		if (typeof togglePortfolioView === "function") togglePortfolioView(isChecked);
	};

	const onWalletClick = () => {
		if (typeof onWalletsDisplay === "function") onWalletsDisplay();
	};

	return (
		<div>
			<div className="mb-4">
				<div className="text-lg text-theme-neutral-800 mb-1">Filter Network</div>
				<div className="text-sm text-theme-neutral-500">Select the types of networks</div>
			</div>
			<SelectNetwork networks={networks} onChange={onNetworkChange} onViewAll={onViewAllNetworks}></SelectNetwork>
			<Divider></Divider>

			<div className="mb-4 flex flex-row">
				<div className="w-4/5">
					<div className="text-lg text-theme-neutral-800 mb-1">Your Wallets</div>
					<div className="text-sm text-theme-neutral-500">Select the display of your wallets</div>
				</div>
				<div className="w-1/5 text-right">
					<div
						data-testid="filter-wallets__wallets"
						className="text-theme-neutral-800 relative mt-3 mr-10 h-6 cursor-pointer"
						onClick={onWalletClick}
					>
						All <Badge icon="ChevronDown" className="top-0 -right-6"></Badge>
					</div>
				</div>
			</div>
			<Divider></Divider>
			<div className="mb-4 flex flex-row">
				<div className="w-4/5">
					<div className="text-lg text-theme-neutral-800 mb-1">Portfolio Chart</div>
					<div className="text-sm text-theme-neutral-500">You an hide this unit if you like</div>
				</div>
				<div className="w-1/5 text-right pr-7 pt-3">
					<Toggle
						checked={showPortfolio}
						data-testid="filter-wallets_toggle--portfolio"
						onChange={(e) => togglePortfolio(e.target.checked)}
					/>
				</div>
			</div>
			<Divider></Divider>
			<div className="mb-4 flex flex-row">
				<div className="w-4/5">
					<div className="text-lg text-theme-neutral-800 mb-1">Transaction History</div>
					<div className="text-sm text-theme-neutral-500">You can hide this unit if you like</div>
				</div>
				<div className="w-1/5 text-right pr-7 pt-3">
					<Toggle
						value={showTransactions}
						data-testid="filter-wallets_toggle--transactions"
						onChange={(e) => toggleTransactions(e.target.checked)}
					/>
				</div>
			</div>
			<Divider></Divider>
		</div>
	);
};

FilterWallets.defaultProps = {
	visibleTransactionsView: false,
	visiblePortfolioView: false,
	networks: [],
};
