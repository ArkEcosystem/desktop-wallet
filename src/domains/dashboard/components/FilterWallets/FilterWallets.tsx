import { Divider } from "app/components//Divider";
import { Badge } from "app/components/Badge";
import { SelectNetwork } from "app/components/SelectNetwork";
import { Toggle } from "app/components/Toggle";
import React, { useState } from "react";

export type FilterWalletsProps = {
	networks?: any;
	visiblePortfolioView?: any;
	visibleTransactionsView?: any;
	onNetworkChange?: any;
	onViewAllNetworks?: any;
	onWalletsDisplay?: any;
	togglePortfolioView?: any;
	toggleTransactionsView?: any;
};

export const FilterWallets = ({
	networks,
	onNetworkChange,
	onViewAllNetworks,
	onWalletsDisplay,
	togglePortfolioView,
	toggleTransactionsView,
	visibleTransactionsView,
	visiblePortfolioView,
}: FilterWalletsProps) => {
	const [showPortfolio, setShowPortfolio] = useState(visiblePortfolioView);
	const [showTransactions, setShowTransactions] = useState(visibleTransactionsView);

	const toggleTransactions = (isChecked: boolean) => {
		setShowTransactions(isChecked);
		if (typeof toggleTransactionsView === "function") toggleTransactionsView(isChecked);
	};

	const togglePortfolio = (isChecked: boolean) => {
		setShowPortfolio(isChecked);
		if (typeof togglePortfolioView === "function") togglePortfolioView(isChecked);
	};

	const onWalletClick = () => {
		if (typeof onWalletsDisplay === "function") onWalletsDisplay();
	};

	return (
		<div className="text-left">
			<div className="mb-4">
				<div className="mb-1 text-sm text-theme-neutral-800">Filter Network</div>
				<div className="text-xs text-theme-neutral-500">Select the types of networks</div>
			</div>
			<SelectNetwork networks={networks} onChange={onNetworkChange} onViewAll={onViewAllNetworks} />
			<Divider />

			<div className="flex flex-row mb-4">
				<div className="w-4/5">
					<div className="mb-1 text-sm text-theme-neutral-800">Your Wallets</div>
					<div className="text-xs text-theme-neutral-500">Select the display of your wallets</div>
				</div>
				<div className="w-1/5 text-right">
					<div
						data-testid="filter-wallets__wallets"
						className="relative h-6 mt-3 mr-5 text-sm cursor-pointer text-theme-neutral-800"
						onClick={onWalletClick}
					>
						All <Badge icon="ChevronDown" className="top-0 -right-6" />
					</div>
				</div>
			</div>
			<Divider />
			<div className="flex flex-row mb-4">
				<div className="w-4/5">
					<div className="mb-1 text-sm text-theme-neutral-800">Portfolio Chart</div>
					<div className="text-xs text-theme-neutral-500">You an hide this unit if you like</div>
				</div>
				<div className="w-1/5 pt-3 text-right">
					<Toggle
						checked={showPortfolio}
						data-testid="filter-wallets_toggle--portfolio"
						onChange={(e) => togglePortfolio(e.target.checked)}
					/>
				</div>
			</div>
			<Divider />
			<div className="flex flex-row mb-4">
				<div className="w-4/5">
					<div className="mb-1 text-sm text-theme-neutral-800">Transaction History</div>
					<div className="text-xs text-theme-neutral-500">You can hide this unit if you like</div>
				</div>
				<div className="w-1/5 pt-3 text-right">
					<Toggle
						checked={showTransactions}
						data-testid="filter-wallets_toggle--transactions"
						onChange={(e) => toggleTransactions(e.target.checked)}
					/>
				</div>
			</div>
		</div>
	);
};

FilterWallets.defaultProps = {
	visibleTransactionsView: false,
	visiblePortfolioView: false,
	networks: [],
};
