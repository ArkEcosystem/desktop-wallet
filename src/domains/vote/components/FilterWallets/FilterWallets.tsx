import { Dropdown, DropdownOption } from "app/components/Dropdown";
import { FilterNetwork } from "app/components/FilterNetwork";
import { Icon } from "app/components/Icon";
import React from "react";
import { useTranslation } from "react-i18next";

export type FilterWalletsProps = {
	networks?: any;
	walletsDisplayType?: string;
	onNetworkChange?: any;
	onViewAllNetworks?: any;
	onWalletsDisplayType?: any;
};

export const FilterWallets = ({
	networks,
	walletsDisplayType,
	onNetworkChange,
	onViewAllNetworks,
	onWalletsDisplayType,
}: FilterWalletsProps) => {
	const { t } = useTranslation();

	const walletDisplayOptions = [
		{ label: t("COMMON.ALL"), value: "all" },
		{ label: t("COMMON.FAVORITES"), value: "favorites" },
		{ label: t("COMMON.LEDGER"), value: "ledger" },
	];

	return (
		<Dropdown
			position="right"
			toggleContent={
				<div className="cursor-pointer">
					<Icon name="Filters" width={20} height={20} />
				</div>
			}
		>
			<div className="px-10 py-7 w-128">
				<div className="flex flex-col text-left" data-testid="FilterWallets">
					<div className="mb-8">
						<div className="font-semibold text-theme-secondary-text">
							{t("DASHBOARD.FILTER_WALLETS.CRYPTOASSET.TITLE")}
						</div>
						<div className="mt-1 text-sm text-theme-neutral">
							{t("DASHBOARD.FILTER_WALLETS.CRYPTOASSET.DESCRIPTION")}
						</div>
					</div>

					<FilterNetwork networks={networks} onChange={onNetworkChange} onViewAll={onViewAllNetworks} />

					<div className="my-8 border-t border-dotted border-theme-neutral-300 dark:border-theme-neutral-800" />

					<div className="flex flex-col">
						<div className="flex items-center justify-between">
							<div className="font-semibold text-theme-secondary-text">
								{t("DASHBOARD.FILTER_WALLETS.WALLETS.TITLE")}
							</div>

							<Dropdown
								toggleIcon="ChevronDown"
								options={walletDisplayOptions}
								onSelect={(option: DropdownOption) => onWalletsDisplayType?.(option)}
								toggleContent={
									<div
										data-testid="filter-wallets__wallets"
										className="flex items-center justify-end cursor-pointer text-theme-secondary-text"
									>
										<span className="inline-block mr-2 font-semibold">
											{
												walletDisplayOptions.find(
													(option) => option.value === walletsDisplayType,
												)?.label
											}
										</span>
										<Icon name="ChevronDown" width={12} height={12} />
									</div>
								}
							/>
						</div>

						<div className="pr-12 mt-1 text-sm text-theme-neutral">
							{t("DASHBOARD.FILTER_WALLETS.WALLETS.DESCRIPTION")}
						</div>
					</div>
				</div>
			</div>
		</Dropdown>
	);
};

FilterWallets.defaultProps = {
	networks: [],
	walletsDisplayType: "all",
};
