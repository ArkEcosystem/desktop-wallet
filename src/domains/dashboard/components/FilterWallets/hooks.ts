import { Profile, ProfileSetting } from "@arkecosystem/platform-sdk-profiles";
import { uniqBy } from "@arkecosystem/utils";
import { useDashboardConfig } from "domains/dashboard/pages/Dashboard/hooks";
import { useMemo } from "react";

import { FilterWalletsHookProps } from "./";

export const useWalletFilters = ({ profile }: { profile: Profile }) => {
	const { defaultConfiguration, setValue, walletsDisplayType, selectedNetworkIds, viewType } = useDashboardConfig({
		profile,
	});

	const allWalletsLength = profile.wallets().values().length;
	const networks = useMemo(() => {
		const networks = profile
			.wallets()
			.values()
			.map((wallet) => ({
				id: wallet.network().id(),
				isLive: wallet.network().isLive(),
				name: wallet.network().name(),
				coin: wallet.network().coin(),
				isSelected: selectedNetworkIds.includes(wallet.network().id()),
			}));

		return uniqBy(networks, (network) => network.id);
	}, [profile, selectedNetworkIds, allWalletsLength]); // eslint-disable-line react-hooks/exhaustive-deps

	const isFilterChanged = useMemo(() => {
		if (walletsDisplayType !== defaultConfiguration.walletsDisplayType) {
			return true;
		}

		if (selectedNetworkIds.length < defaultConfiguration.selectedNetworkIds.length) {
			return true;
		}

		return false;
	}, [walletsDisplayType, selectedNetworkIds, defaultConfiguration]);

	return useMemo<FilterWalletsHookProps & { update: (key: string, value: any) => void }>(
		() => ({
			networks,
			useTestNetworks: profile.settings().get(ProfileSetting.UseTestNetworks),
			walletsDisplayType,
			selectedNetworkIds,
			isFilterChanged,
			viewType,
			update: setValue,
		}),
		[walletsDisplayType, selectedNetworkIds, viewType, isFilterChanged, networks, profile, setValue],
	);
};
