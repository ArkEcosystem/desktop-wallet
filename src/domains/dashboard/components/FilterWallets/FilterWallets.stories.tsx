import React from "react";
import { FilterWallets } from "./FilterWallets";

export default {
	title: "Dashboard / Components / FilterWallets",
};

export const Default = () => {
	const networks = [
		{
			name: "Ark",
			isSelected: true,
		},
		{
			name: "Eth",
			isSelected: true,
		},
		{
			name: "Btc",
			isSelected: false,
		},
	];

	let visiblePortfolioView = true;
	let visibleTransactionsView = true;

	return (
		<div className="w-128">
			<FilterWallets
				visibleTransactionsView={visibleTransactionsView}
				visiblePortfolioView={visiblePortfolioView}
				networks={networks}
				onNetworkChange={(changedNetwork: any, newNetworksList: any) => {
					console.log("changed network", changedNetwork);
					console.log("changed network new list", newNetworksList);
				}}
				togglePortfolioView={(isChecked: boolean) => (visiblePortfolioView = isChecked)}
				toggleTransactionsView={(isChecked: boolean) => (visibleTransactionsView = isChecked)}
				onWalletsDisplay={() => alert("Display wallets  event")}
				onViewAllNetworks={() => alert("View all networks")}
			></FilterWallets>
		</div>
	);
};
