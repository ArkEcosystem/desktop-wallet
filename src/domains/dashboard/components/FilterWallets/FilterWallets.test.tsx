import { act } from "@testing-library/react-hooks";
import React from "react";
import { fireEvent, render } from "testing-library";

import { FilterWallets } from "./FilterWallets";

const networks = [
	{
		name: "Ark",
		isSelected: true,
	},
	{
		name: "Ethereum",
		isSelected: true,
	},
	{
		name: "Bitcoin",
		isSelected: false,
	},
];

describe("FilterWallets", () => {
	it("should render", () => {
		const { container } = render(<FilterWallets />);
		expect(container).toMatchSnapshot();
	});

	it("should render with networks selection", () => {
		const { container } = render(<FilterWallets networks={networks} />);
		expect(container).toMatchSnapshot();
	});

	it("should emit event for transactions view toggle", () => {
		const fn = jest.fn();
		const { getByTestId } = render(<FilterWallets networks={networks} toggleTransactionsView={fn} />);
		const toggle = getByTestId("filter-wallets_toggle--transactions");

		act(() => {
			fireEvent.click(toggle);
		});
		expect(fn).toBeCalled();
	});

	it("should ignore emit event for transactions view if callback not provided", () => {
		const fn = jest.fn();
		const { getByTestId } = render(<FilterWallets networks={networks} />);
		const toggle = getByTestId("filter-wallets_toggle--transactions");

		act(() => {
			fireEvent.click(toggle);
		});
		expect(fn).not.toBeCalled();
	});

	it("should emit event for portfolio view toggle", () => {
		const onPortfolioView = jest.fn();
		const { getByTestId } = render(<FilterWallets networks={networks} togglePortfolioView={onPortfolioView} />);
		const toggle = getByTestId("filter-wallets_toggle--portfolio");

		act(() => {
			fireEvent.click(toggle);
		});
		expect(onPortfolioView).toBeCalled();
	});

	it("should ignore emit event for portfolio view if callback not provided", () => {
		const fn = jest.fn();
		const { getByTestId } = render(<FilterWallets networks={networks} />);
		const toggle = getByTestId("filter-wallets_toggle--portfolio");

		act(() => {
			fireEvent.click(toggle);
		});
		expect(fn).not.toBeCalled();
	});

	it("should emit event for wallets display", () => {
		const fn = jest.fn();
		const { getByTestId } = render(<FilterWallets networks={networks} onWalletsDisplay={fn} />);

		act(() => {
			fireEvent.click(getByTestId("filter-wallets__wallets"));
		});

		act(() => {
			fireEvent.click(getByTestId("dropdown__option--0"));
		});

		expect(fn).toBeCalled();
	});

	it("should ignore emit event for wallet display if callback not provided", () => {
		const fn = jest.fn();
		const { getByTestId } = render(<FilterWallets networks={networks} />);
		act(() => {
			fireEvent.click(getByTestId("filter-wallets__wallets"));
		});

		act(() => {
			fireEvent.click(getByTestId("dropdown__option--0"));
		});

		expect(fn).not.toBeCalled();
	});
});
