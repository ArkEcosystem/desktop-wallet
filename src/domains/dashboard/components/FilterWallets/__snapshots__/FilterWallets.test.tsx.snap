// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterWallets should render 1`] = `
<div>
  <div
    class="text-left"
  >
    <div
      class="mb-8"
    >
      <div
        class="mb-3 font-semibold text-theme-neutral-700"
      >
        Filter Network
      </div>
      <div
        class="text-sm text-theme-neutral-500"
      >
        Select the types of networks
      </div>
    </div>
    <div>
      <ul
        class="inline-block"
        data-testid="network__option"
      />
      <div
        class="sc-AxiKw jQtkkF relative ml-2 cursor-pointer border-theme-primary-100"
        data-testid="network__viewall"
      >
        <div
          class="text-sm font-semibold text-theme-primary-500"
        >
          All
        </div>
        <span
          class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent border-theme-primary-100 -bottom-1 -right-4 text-theme-primary-500"
        >
          <div
            class="sc-AxjAm iepfoc"
            height="10"
            width="10"
          >
            <svg>
              chevron-down.svg
            </svg>
          </div>
          <span />
        </span>
      </div>
    </div>
    <div
      class="my-8 border-t border-dotted border-1 border-theme-neutral-200"
    />
    <div
      class="flex flex-row mb-8"
    >
      <div
        class="w-4/5"
      >
        <div
          class="mb-3 font-semibold text-theme-neutral-700"
        >
          Your Wallets
        </div>
        <div
          class="text-sm text-theme-neutral-500"
        >
          Select the display of your wallets
        </div>
      </div>
      <div
        class="w-1/5 text-right"
      >
        <div
          class="relative h-6 mt-3 mr-5 cursor-pointer text-theme-neutral-700"
          data-testid="filter-wallets__wallets"
        >
          <span
            class="inline-block pr-1 font-semibold"
          >
            All
          </span>
          <span
            class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent top-1 -right-6"
          >
            <div
              class="sc-AxjAm pXfjM"
              height="15"
              width="15"
            >
              <svg>
                chevron-down.svg
              </svg>
            </div>
            <span />
          </span>
        </div>
      </div>
    </div>
    <div
      class="my-8 border-t border-dotted border-1 border-theme-neutral-200"
    />
    <div
      class="flex flex-row mb-6"
    >
      <div
        class="w-4/5"
      >
        <div
          class="mb-3 font-semibold text-theme-neutral-700"
        >
          Portfolio Chart
        </div>
        <div
          class="text-sm text-theme-neutral-500"
        >
          You an hide this unit if you like
        </div>
      </div>
      <div
        class="w-1/5 pt-3 text-right"
      >
        <label>
          <input
            class="sc-AxhCb crfxJX"
            data-testid="filter-wallets_toggle--portfolio"
            type="checkbox"
          />
          <div
            class="sc-AxhUy fdlXZX"
          >
            <span
              class="sc-AxgMl jilcKf"
            />
          </div>
        </label>
      </div>
    </div>
    <div
      class="flex flex-row mb-4"
    >
      <div
        class="w-4/5"
      >
        <div
          class="mb-3 font-semibold text-theme-neutral-700"
        >
          Transaction History
        </div>
        <div
          class="text-sm text-theme-neutral-500"
        >
          You can hide this unit if you like
        </div>
      </div>
      <div
        class="w-1/5 pt-3 text-right"
      >
        <label>
          <input
            class="sc-AxhCb crfxJX"
            data-testid="filter-wallets_toggle--transactions"
            type="checkbox"
          />
          <div
            class="sc-AxhUy fdlXZX"
          >
            <span
              class="sc-AxgMl jilcKf"
            />
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`FilterWallets should render with networks selection 1`] = `
<div>
  <div
    class="text-left"
  >
    <div
      class="mb-8"
    >
      <div
        class="mb-3 font-semibold text-theme-neutral-700"
      >
        Filter Network
      </div>
      <div
        class="text-sm text-theme-neutral-500"
      >
        Select the types of networks
      </div>
    </div>
    <div>
      <ul
        class="inline-block"
        data-testid="network__option"
      >
        <li
          class="inline-block mr-5 cursor-pointer"
          data-testid="network__option--0"
        >
          <div
            class="sc-AxiKw jQtkkF relative border-theme-success-500 text-theme-success-500"
          >
            <div
              class="sc-AxjAm buCJuk"
              height="20"
              width="20"
            >
              <svg>
                ark.svg
              </svg>
            </div>
            <span
              class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent -bottom-1 -right-4 bg-theme-success-500 text-theme-success-contrast"
            >
              <div
                class="sc-AxjAm HWmvM"
                height="12"
                width="12"
              >
                <svg>
                  checkmark.svg
                </svg>
              </div>
              <span />
            </span>
          </div>
        </li>
        <li
          class="inline-block mr-5 cursor-pointer"
          data-testid="network__option--1"
        >
          <div
            class="sc-AxiKw jQtkkF relative border-theme-success-500 text-theme-success-500"
          >
            <div
              class="sc-AxjAm buCJuk"
              height="20"
              width="20"
            >
              <svg>
                eth.svg
              </svg>
            </div>
            <span
              class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent -bottom-1 -right-4 bg-theme-success-500 text-theme-success-contrast"
            >
              <div
                class="sc-AxjAm HWmvM"
                height="12"
                width="12"
              >
                <svg>
                  checkmark.svg
                </svg>
              </div>
              <span />
            </span>
          </div>
        </li>
        <li
          class="inline-block mr-5 cursor-pointer"
          data-testid="network__option--2"
        >
          <div
            class="sc-AxiKw jQtkkF relative border-theme-neutral-200 text-theme-neutral-300"
          >
            <div
              class="sc-AxjAm buCJuk"
              height="20"
              width="20"
            >
              <svg>
                btc.svg
              </svg>
            </div>
            <span
              class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent border-theme-neutral-200 -bottom-1 -right-4"
            >
              <span />
            </span>
          </div>
        </li>
      </ul>
      <div
        class="sc-AxiKw jQtkkF relative ml-2 cursor-pointer border-theme-primary-100"
        data-testid="network__viewall"
      >
        <div
          class="text-sm font-semibold text-theme-primary-500"
        >
          All
        </div>
        <span
          class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent border-theme-primary-100 -bottom-1 -right-4 text-theme-primary-500"
        >
          <div
            class="sc-AxjAm iepfoc"
            height="10"
            width="10"
          >
            <svg>
              chevron-down.svg
            </svg>
          </div>
          <span />
        </span>
      </div>
    </div>
    <div
      class="my-8 border-t border-dotted border-1 border-theme-neutral-200"
    />
    <div
      class="flex flex-row mb-8"
    >
      <div
        class="w-4/5"
      >
        <div
          class="mb-3 font-semibold text-theme-neutral-700"
        >
          Your Wallets
        </div>
        <div
          class="text-sm text-theme-neutral-500"
        >
          Select the display of your wallets
        </div>
      </div>
      <div
        class="w-1/5 text-right"
      >
        <div
          class="relative h-6 mt-3 mr-5 cursor-pointer text-theme-neutral-700"
          data-testid="filter-wallets__wallets"
        >
          <span
            class="inline-block pr-1 font-semibold"
          >
            All
          </span>
          <span
            class="sc-AxirZ nnMcM flex border-2 rounded-full justify-center w-5 h-5 items-center align-middle absolute bg-theme-background border-transparent top-1 -right-6"
          >
            <div
              class="sc-AxjAm pXfjM"
              height="15"
              width="15"
            >
              <svg>
                chevron-down.svg
              </svg>
            </div>
            <span />
          </span>
        </div>
      </div>
    </div>
    <div
      class="my-8 border-t border-dotted border-1 border-theme-neutral-200"
    />
    <div
      class="flex flex-row mb-6"
    >
      <div
        class="w-4/5"
      >
        <div
          class="mb-3 font-semibold text-theme-neutral-700"
        >
          Portfolio Chart
        </div>
        <div
          class="text-sm text-theme-neutral-500"
        >
          You an hide this unit if you like
        </div>
      </div>
      <div
        class="w-1/5 pt-3 text-right"
      >
        <label>
          <input
            class="sc-AxhCb crfxJX"
            data-testid="filter-wallets_toggle--portfolio"
            type="checkbox"
          />
          <div
            class="sc-AxhUy fdlXZX"
          >
            <span
              class="sc-AxgMl jilcKf"
            />
          </div>
        </label>
      </div>
    </div>
    <div
      class="flex flex-row mb-4"
    >
      <div
        class="w-4/5"
      >
        <div
          class="mb-3 font-semibold text-theme-neutral-700"
        >
          Transaction History
        </div>
        <div
          class="text-sm text-theme-neutral-500"
        >
          You can hide this unit if you like
        </div>
      </div>
      <div
        class="w-1/5 pt-3 text-right"
      >
        <label>
          <input
            class="sc-AxhCb crfxJX"
            data-testid="filter-wallets_toggle--transactions"
            type="checkbox"
          />
          <div
            class="sc-AxhUy fdlXZX"
          >
            <span
              class="sc-AxgMl jilcKf"
            />
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;
